src/v/raft/tests/raft_fixture.h

```diff
@@ -211,6 +214,17 @@ class raft_node_instance : public ss::weakly_referencable<raft_node_instance> {
 
     ss::future<model::offset> random_batch_base_offset(model::offset max);
 
+    /// \brief Sets a callback function to be invoked when the leader dispatches
+    /// a message to followers.
+    ///
+    /// It is invoked once for each follower. The dispatch process will proceed
+    /// once the returned future from the callback function is resolved.
+    ///
+    /// This method is handy for failure injection.
+    ///
+    //// \param f The callback function to be invoked when a message is
+    /// dispatched.
+    void on_dispatch(ss::noncopyable_function<ss::future<>(msg_type)> f);
```

Comment:
idea: perhaps we could add the node_id to which we are dispatching as another argument to the injected function. It allows for more granularity. Doesn't have to happen in this PR though.

Reply:
@VladLazar yep. Thought about that too. Would a struct be even better to avoid having to refactor all places if we want to propagate more info down the road? Having node_id/to_node_id has an added benefit of communicating at what level dispatching is happening. Ie that it’s happening on dispatching to the follower rather than somewhere else up the stack.

Reply:
Yeah. A struct would be better than two args.